setup.py

```diff
@@ -74,6 +77,19 @@ def clean_version(v):
       setup_requires=[],
       )
 
+if current_platform != "windows":
+    bash_file = "~/.bash_profile" if current_platform == "darwin" else "~/.bashrc"
+    argcomplete_command = 'eval "$(register-python-argcomplete retriever)"'
+    with open(os.path.expanduser(bash_file), "a+") as bashrc:
+        bashrc.seek(0)
+        # register retriever for arg-completion if not already registered
+        # whenever a new shell is spawned
+        if argcomplete_command not in bashrc.read():
+            bashrc.write(argcomplete_command + "\n")
+            bashrc.close()
+    os.system("activate-global-python-argcomplete")
```

Comment:
I don't think its of much use in windows. Although I haven't tested in windows, judging from the output of this line,
```bash
Installing bash completion script /usr/local/etc/bash_completion.d/python-argcomplete.sh
```
- I would say it wont work in windows environment.